pkg/ccl/changefeedccl/alter_changefeed_stmt.go

```diff
@@ -174,21 +175,44 @@ func alterChangefeedPlanHook(
 				}
 
 				for key, value := range opts {
-					if _, ok := changefeedbase.ChangefeedOptionExpectValues[key]; !ok {
-						return pgerror.Newf(pgcode.InvalidParameterValue, `invalid option %q`, key)
-					}
 					if _, ok := changefeedbase.AlterChangefeedUnsupportedOptions[key]; ok {
 						return pgerror.Newf(pgcode.InvalidParameterValue, `cannot alter option %q`, key)
 					}
-					opt := tree.KVOption{Key: tree.Name(key)}
-					if len(value) > 0 {
-						opt.Value = tree.NewDString(value)
+					if key == changefeedbase.OptSink {
+						newSinkURI, err := url.Parse(value)
+						if err != nil {
+							return err
+						}
+
+						prevSinkURI, err := url.Parse(prevDetails.SinkURI)
+						if err != nil {
+							return err
+						}
+
+						if newSinkURI.Scheme != prevSinkURI.Scheme {
+							return pgerror.Newf(
+								pgcode.InvalidParameterValue,
+								`new sink type %q does not match original sink type %q, sink type cannot be altered`,
```

Comment:
It’d be nice to make it a little more prescriptive, so the user knows what to do next. something like ‘changing the type of a changefeed (in this case %q to %q) is disallowed. use CREATE CHANGEFEED instead.'